rfcs/text/0011_persistable_state_service.md

```diff
@@ -0,0 +1,218 @@
+- Start Date: 2020-06-02
+- RFC PR: (leave this empty)
+- Kibana Issue: (leave this empty)
+
+# Summary
+
+Pluggable state from registry items can often end up stored inside saved objects, for instance:
+Embeddables, Expression strings, Url generators, ...
+
+When plugin A (persister) stores some state that belongs to another plugin B a few issues arise:
+- How does persister know if plugin B state contains references to saved objects
+- How does persister migrate the saved object when it contains state that belongs to plugin B
+- How does persister know if state that belongs to plugin B contains state that belongs to plugin C
+
+# Basic example
```

Comment:
nit: I'm not sure I have any great suggestions on this, but the basic example still seems pretty verbose and takes a few minutes to understand; I wonder if there's a way to simplify it even further so that folks can immediately grasp the idea.

Or perhaps add a few sentences to the summary to better set up what you are going to do?

Reply:
will add a few sentences. i had a more basic example here but we extended it to better show what you are exposed to do.